infra/infra.go

```diff
@@ -126,11 +129,16 @@ type NodePool interface {
 
 // Node defines an interface to a remote node
 type Node interface {
-	// Addr returns the address of the node
+	// Addr returns public address of the node
 	Addr() string
+	// PrivateAddr returns the private address of the node
+	PrivateAddr() string
 	// Connect connects to this node and returns a new session object
 	// that can be used to execute remote commands
 	Connect() (*ssh.Session, error)
```

Comment:
We don't need `Connect` as part of `Node` anymore - one can always create a `Session` from a `Client`. Besides, the implementation is identical and hence duplicate in each provisioner.

I think internal unit-tests are out-of-date w.r.t changes to this interface.